src/core/application/useCases/pedido/index.ts

```diff
@@ -1,15 +1,55 @@
+import { Pedido, StatusPedidoEnum } from "@domain/entities/pedido";
 import { IPedidoUseCase } from "./pedido.interface";
-import { Pedido } from "@domain/entities/pedido";
 import { AssertionConcern } from "@domain/base/assertionConcern";
-import { IPedidoRepository } from "@domain/repositories/pedidoRepository.interface";
+import { PedidoRepository } from "@domain/repositories/pedidoRepository.interface";
+import { ProdutoRepository } from "@domain/repositories/produtoRepository.interface";
+import { ValorTotal } from "@domain/valueObjects/valorTotal";
 
 export class PedidoUseCase implements IPedidoUseCase {
-    constructor(private readonly pedidoRepository: IPedidoRepository) {}
+    private readonly pedidoRepository: PedidoRepository;
+    private readonly produtoRepository: ProdutoRepository;
+
+    constructor(
+        pedidoRepository: PedidoRepository,
+        produtoRepository: ProdutoRepository,
+    ) {
+        this.pedidoRepository = pedidoRepository;
+        this.produtoRepository = produtoRepository;
+    }
 
     getAll(filters?: Partial<Pedido>): Promise<Pedido[]> {
         return this.pedidoRepository.getAll(filters);
     }
 
+    public async getById(id: string): Promise<Pedido> {
+        return this.pedidoRepository.getById(id);
+    }
+
+    public async create(pedido: Pedido): Promise<Pedido> {
+        const ids = pedido.itens.map((item) => item.produtoId);
+
+        const itensPedido = await this.produtoRepository.getByIds(ids);
+
+        const itensComPreco = pedido.itens.map((item) => ({
+            ...item,
+            preco: itensPedido.find((produto) => produto.id === item.produtoId)
+                .preco,
+        }));
+
+        const valorTotal = ValorTotal.create(itensComPreco);
+
+        if (pedido.status && pedido.status !== StatusPedidoEnum.Recebido) {
+            throw new Error("Não é necessário informar o status");
+        }
+
+        return this.pedidoRepository.create(
+            new Pedido({
+                ...pedido,
+                valorTotal: valorTotal.value,
+            }),
+        );
+    }
+
     async update(
```

Comment:
micro detalhe, mas aqui pra manter a consistência eu adicionaria o public
```suggestion
    public async update(
```